### PR TITLE
Rule Updates to remove misleading 'all' usage

### PR DESCRIPTION
**Pull request by morriscode**

### CODE REVIEW

File: detection-rules/link_credential_phishing.yml [hunk 1]
```diff
@@ -6,11 +6,9 @@ severity: "high"
 source: |
   type.inbound 
   and any(body.links, 
-      all([beta.linkanalysis(.)],
-          .credphish.disposition == "phishing"
-          and .credphish.brand.confidence in ("medium", "high")
-      )
-  ) 
+      beta.linkanalysis(.).credphish.disposition == "phishing"
+          and beta.linkanalysis(.).credphish.brand.confidence in ("medium", "high")
```
Comment (rw-access) [inline detection-rules/link_credential_phishing.yml:10]:
```suggestion
      and beta.linkanalysis(.).credphish.brand.confidence in ("medium", "high")
```

File: detection-rules/link_login_or_captcha.yml [hunk 2]
```diff
@@ -6,15 +6,13 @@ severity: "high"
...
   and any(body.links, 
-      all([beta.linkanalysis(.)], (
-          .credphish.contains_login
-          or .credphish.contains_captcha
+      beta.linkanalysis(.).credphish.contains_login
+          or beta.linkanalysis(.).credphish.contains_captcha
+      )
+      and any(body.links,
+          beta.linkanalysis(.).effective_url.domain.root_domain in $free_subdomain_hosts
+          or .href_url.domain.root_domain in $free_subdomain_hosts
```
Comment (rw-access) [inline detection-rules/link_login_or_captcha.yml:14]:
```suggestion
      beta.linkanalysis(.).credphish.contains_login
      or beta.linkanalysis(.).credphish.contains_captcha
  )
  and any(body.links,
          beta.linkanalysis(.).effective_url.domain.root_domain in $free_subdomain_hosts
          or .href_url.domain.root_domain in $free_subdomain_hosts
  )
```

**Review by rw-access — COMMENTED** [review]
looks like some indentation needs to be fixed, and i think there was a missing `)`

File: detection-rules/link_credential_phishing.yml (hunk 1 above)
Comment (morriscode) [reply detection-rules/link_credential_phishing.yml:10]: Thanks!! 👍  I didn't see a missing paren, both files didn't surface any syntax highlights in the editor, but it's entirely possible :)

File: detection-rules/link_login_or_captcha.yml (hunk 2 above)
Comment (morriscode) [reply detection-rules/link_login_or_captcha.yml]: Actually the additional paren inclusion now throws a syntax highlight..

File: detection-rules/link_login_or_captcha.yml [hunk 3]
```diff
@@ -6,15 +6,14 @@ severity: "high"
...
-          or ..href_url.domain.root_domain in $free_subdomain_hosts
-      ))
+      beta.linkanalysis(.).credphish.contains_login
+      or beta.linkanalysis(.).credphish.contains_captcha
+  )
+  and any(body.links,
+          beta.linkanalysis(.).effective_url.domain.root_domain in $free_subdomain_hosts
+          or .href_url.domain.root_domain in $free_subdomain_hosts
   )
+      )
```
Comment (rw-access) [inline detection-rules/link_login_or_captcha.yml:16]:
```suggestion
```

File: detection-rules/link_login_or_captcha.yml [hunk 4]
```diff
@@ -6,15 +6,14 @@ severity: "high"
...
-      and (
-          .effective_url.domain.root_domain in $free_subdomain_hosts
-          or ..href_url.domain.root_domain in $free_subdomain_hosts
-      ))
+      beta.linkanalysis(.).credphish.contains_login
+      or beta.linkanalysis(.).credphish.contains_captcha
+  )
+  and any(body.links,
+          beta.linkanalysis(.).effective_url.domain.root_domain in $free_subdomain_hosts
+          or .href_url.domain.root_domain in $free_subdomain_hosts
```
Comment (rw-access) [inline detection-rules/link_login_or_captcha.yml:14]: got changed on these too, idk if my suggestions didn't work or something
```suggestion
      beta.linkanalysis(.).effective_url.domain.root_domain in $free_subdomain_hosts
      or .href_url.domain.root_domain in $free_subdomain_hosts
```

File: detection-rules/link_login_or_captcha.yml [hunk 5]
```diff
@@ -6,14 +6,12 @@ severity: "high"
 source: |
   type.inbound 
   and any(body.links, 
-      all([beta.linkanalysis(.)], (
-          .credphish.contains_login
-          or .credphish.contains_captcha
-      )
-      and (
-          .effective_url.domain.root_domain in $free_subdomain_hosts
```
Comment (rw-access) [inline detection-rules/link_login_or_captcha.yml:14]: woops, i must've clobbered over these! we still want this check right?

Comment (morriscode) [reply detection-rules/link_login_or_captcha.yml:14]: yessir

**Comment by morriscode** [comment]
Closing to split changes

**Pull request closed without merging**